modular_skyrat/modules/death_consequences_perk/death_consequences_trauma.dm

```diff
@@ -0,0 +1,525 @@
+	if (owner.status_flags & GODMODE)
+		return
+
+	var/is_dead = (owner.stat == DEAD)
+	var/degradation_increase = get_passive_degradation_increase(is_dead) * seconds_per_tick
+	var/degradation_reduction = get_passive_degradation_decrease(is_dead) * seconds_per_tick
+
+	adjust_degradation(degradation_increase - degradation_reduction)
+
+	// Ensure our victim's stamina is at or above our minimum stamina damage
+	if (!is_dead)
+		damage_stamina(seconds_per_tick)
+
+	if ((world.time - time_between_reminders) > time_of_last_message_sent)
+		send_reminder()
+
+/datum/brain_trauma/severe/death_consequences/proc/get_passive_degradation_increase(is_dead)
+	var/increase = 0
+
+	if (is_dead)
+		increase += base_degradation_per_second_while_dead
+
+		if (owner.has_reagent(/datum/reagent/toxin/formaldehyde, needs_metabolizing = FALSE))
+			var/datum/reagent/reagent_instance = owner.reagents.get_reagent(/datum/reagent/toxin/formaldehyde)
+			if (reagent_process_flags_valid(owner, reagent_instance))
+				increase *= formaldehyde_death_degradation_mult
+	else
+		if (base_degradation_reduction_per_second_while_alive < 0) // if you wanna die slowly while alive, go ahead bud
+			increase -= base_degradation_reduction_per_second_while_alive
+
+	if (IS_IN_STASIS(owner))
+		increase *= on_stasis_death_degradation_mult
+
+	return increase
+
+/datum/brain_trauma/severe/death_consequences/proc/get_passive_degradation_decrease(is_dead)
+	var/decrease = 0
+
+	if (!is_dead)
+		if (base_degradation_reduction_per_second_while_alive > 0)
+			decrease += base_degradation_reduction_per_second_while_alive
+
+		if (owner.has_reagent(/datum/reagent/medicine/rezadone, needs_metabolizing = TRUE))
+			var/datum/reagent/reagent_instance = owner.reagents.get_reagent(/datum/reagent/medicine/rezadone)
+			if ((reagent_instance.purity >= rezadone_minimum_purity) && reagent_process_flags_valid(owner, reagent_instance))
+				decrease += rezadone_degradation_decrease
+
+		if (owner.has_reagent(/datum/reagent/medicine/sansufentanyl, needs_metabolizing = TRUE))
+			var/datum/reagent/reagent_instance = owner.reagents.get_reagent(/datum/reagent/medicine/sansufentanyl)
+			if (reagent_process_flags_valid(owner, reagent_instance))
+				decrease += sansufentanyl_degradation_decrease
+
+	if (owner.has_reagent(/datum/reagent/eigenstate))
+		decrease += eigenstasium_degradation_decrease
+
+	if (owner.has_reagent(/datum/reagent/medicine/strange_reagent, needs_metabolizing = TRUE))
+		var/datum/reagent/reagent_instance = owner.reagents.get_reagent(/datum/reagent/medicine/strange_reagent)
+		if (reagent_process_flags_valid(owner, reagent_instance))
+			decrease += strange_reagent_degradation_decrease
+
+	return (decrease * get_passive_degradation_decrease_mult())
+
+#define DEGRADATION_REDUCTION_SLEEPING_MULT 3
+#define DEGRADATION_REDUCTION_RESTING_MULT 1.5
+/// A global proc used for all scenarios we would decrease passive degradation.
```

Comment:
```suggestion

/// A global proc used for all scenarios we would decrease passive degradation.
```

Reply:
Also that documentation isn't true, this isn't a global proc.

Reply:
dunno why i said that